floodviz/maputils.py

```diff
@@ -0,0 +1,51 @@
+import requests
+
+# function to generate a dictionary of site data after querying NWIS
+def site_dict(site_list, url_prefix):
+    # generate the string of site ids for the url
+    id_input_string = ",".join(site_list)
+
+    # create the url
+    url = url_prefix + "/?format=rdb&sites=" + id_input_string + "&siteStatus=all"
+
+    # get data from url
+    data = ""
+    req = requests.get(url)
+    for line in req.text.splitlines():
+        if not line.startswith("#"):
+            data += line + '\n'
+
+    # create array by splitting lines
+    data = data.split('\n')
+
+    # make a list of dicts  from data
+    fields = data[0].split('\t')
+    dnice = []
+    for line in data[2:]:
+        line = line.split('\t')
+        line_dict = dict(zip(fields, line))
+        dnice.append(line_dict)
+
+    # remove bad element from end of list
```

Comment:
What is the bad element? I've written rdb parsers and I don't think I had to do this.